File: core/src/main/java/pcb/uwu/unit/derived/fundamental/VolumeUnit.java
package pcb.uwu.unit.derived.fundamental;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.unit.base.LengthUnit;

public abstract class VolumeUnit extends CompositeUnit {

	public VolumeUnit(LengthUnit length, LengthUnit height, LengthUnit depth) {
		super(new UnitCounter()
				.major(length)
				.major(height)
				.major(depth));
	}
}

File: core/src/main/java/pcb/uwu/unit/derived/fundamental/VolumeUnit.kt
package pcb.uwu.unit.derived.fundamental

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit

open class VolumeUnit : CompositeUnit
{
    constructor(length: LengthUnit,
                height: LengthUnit,
                depth: LengthUnit)
            : super(UnitCounter()
                        .major(length)
                        .major(height)
                        .major(depth))

    constructor(areaUnit: AreaUnit,
                length: LengthUnit)
            : super(areaUnit.unitCounter.major(length))
}